google-cloud-pubsub/lib/google/cloud/pubsub/service.rb

```diff
@@ -204,18 +204,14 @@ def list_subscriptions options = {}
         ##
         # Creates a subscription on a given topic for a given subscriber.
         def create_subscription topic, subscription_name, options = {}
-          subscriber.create_subscription \
-            name:                       subscription_path(subscription_name, options),
-            topic:                      topic_path(topic),
-            push_config:                options[:push_config],
-            ack_deadline_seconds:       options[:deadline],
-            retain_acked_messages:      options[:retain_acked],
-            message_retention_duration: Convert.number_to_duration(options[:retention]),
-            labels:                     options[:labels],
-            enable_message_ordering:    options[:message_ordering],
-            filter:                     options[:filter],
-            dead_letter_policy:         dead_letter_policy(options),
-            retry_policy:               options[:retry_policy]
+          options[:name] = subscription_path subscription_name, options
+          options[:topic] = topic_path topic
+          options[:message_retention_duration] = Convert.number_to_duration options.delete(:retention)
+          options[:dead_letter_policy] = dead_letter_policy options
+          options[:ack_deadline_seconds] = options.delete :deadline
+          options[:retain_acked_messages] = options.delete :retain_acked
+          options[:enable_message_ordering] = options.delete :message_ordering
+          subscriber.create_subscription(**options.compact)
```

Comment:
Have tried updating this to construct a new options using the given options rather than editing in place. This removes the need of having unclean deletes in place. Please let me know your thought.